crates/hir-ty/benches/incremental.rs

```diff
@@ -0,0 +1,212 @@
+use base_db::{
+    CrateGraphBuilder, CratesMap, FileSourceRootInput, FileText, RootQueryDb, SourceDatabase,
+    SourceRoot, SourceRootId, SourceRootInput,
+};
+use criterion::{Criterion, black_box, criterion_group, criterion_main};
+use hir_def::{DefDatabase, ModuleId, nameres::crate_def_map};
+use hir_ty::db::HirDatabase;
+use salsa::Durability;
+use span::FileId;
+use std::{fmt, panic};
+use test_fixture::WithFixture;
+use triomphe::Arc;
+
+#[salsa_macros::db]
+#[derive(Clone)]
+pub(crate) struct BenchmarkDB {
```

Comment:
Maybe we need to do something with all the boilerplate with declaring a db.

Reply:
Can't we reuse the `TestDB`? Why do we need a new type